### ISSUE TITLE
ESPHome missing support for important device_class

### ISSUE THREAD

**Issue by DrCoolzic** [esphome/issues#3405]
### The problem

Currently ESPHome does not support two very important device classes: “apparent_power” and “reactive_power”. This is very annoying because currently **there is no way for a sensor to report values with VA or VAR** for unit_of_measurement. 
If using device_class=power these units of measurement are rejected for statistical data recording …

In the Sensor Component documentation, it is mentioned that:

> device_class (Optional, string): The device class for the sensor. See https://developers.home-assistant.io/docs/core/entity/sensor/#available-device-classes for a list of available options.

But as mentioned two are missing: apparent_power and reactive_power

### Which version of ESPHome has the issue?

2022.6.2

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.6.7

### What platform are you using?

ESP8266

### Board

irrelevant

### Component causing the issue

irrelevant

### Example YAML snippet

```yaml
- platform: teleinfo
    tag_name: "SINSTS"
    name: "linky_sinsts"
    device_class: "apparent_power"
    state_class: "measurement"
    unit_of_measurement: "VA"
```

### Anything in the logs that might be useful for us?

```txt
sensor.teleinfo: [source /config/esphome/linky-tic.yaml:161]
  platform: teleinfo
  tag_name: SINSTS
  name: linky_sinsts
  
  Unknown value 'apparent_power', valid options are '', 'aqi', 'battery', 'carbon_dioxide', 'carbon_monoxide', 'current', 'duration', 'energy', 'gas', 'humidity', 'illuminance', 'monetary', 'nitrogen_dioxide', 'nitrogen_monoxide', 'nitrous_oxide', 'ozone', 'pm1', 'pm10', 'pm25', 'power', 'power_factor', 'pressure', 'signal_strength', 'sulphur_dioxide', 'temperature', 'timestamp', 'volatile_organic_compounds', 'voltage'.
  device_class: apparent_power
  state_class: measurement
  unit_of_measurement: VA
sensor.teleinfo: [source /config/esphome/linky-tic.yaml:168]
  platform: teleinfo
  tag_name: SINSTS1
  name: linky_sinsts1
```

### Additional information

If trying to use power instead of apparent_power
![image](https://user-images.githubusercontent.com/6078364/175781997-f6205ef4-a02d-4199-8624-7fe96f0dbfc8.png)

we get error
![image](https://user-images.githubusercontent.com/6078364/175782058-a8cefa36-d908-49c4-8b44-316beb25a319.png)

Please refer to
Add sensor device classes for apparent and reactive power #688 https://github.com/home-assistant/architecture/discussions/688

and 
Device class “power” should support “VA” and “VAR” units #57236 https://github.com/home-assistant/core/issues/57236

for more info

**Comment by DrCoolzic** [comment]
Hey there anybody looking at this problem?
Badly needed for the Teleinfo component!
Thanks

**Comment by DrCoolzic** [comment]
Can someone look at this problem. It should not be a big modification to add the two missing device classes :)

**Comment by jesserockz** [comment]
These were added to `dev` in https://github.com/esphome/esphome/pull/3624

**Comment by DrCoolzic** [comment]
Many thanks. 
Is this update available on 2022.6.3 ?
